File: Wacomi.SPA/src/app/_models/Attraction.ts
import { AppUser } from "./AppUser";
import { Photo } from "./Photo";
import { Category } from "./Category";
import { AttractionReview } from "./AttractionReview";

export interface Attraction {
    id: number;
    appUser: AppUser;
    appUserId?: number;
    cityName: string;
    cityId: number;
    name: string;
    introduction: string;
    accessInfo: string;
    latitude: number;
    longitude: number;
    photos: Photo[];
    reviewPhotos: Photo[];
    scoreAverage: number;
    likedCount: number;
    isLiked : boolean;
    reviewedCount: boolean;
    attractionReviews: AttractionReview[];
    categories: Category[];
    mainPhotoId?: number;
    mainPhotoUrl?: string;
    websiteUrl?: string;
    dateCreated: Date;
    dateUpdated: Date;
}
// public int Id {get; set;}
//         public AppUserForReturnDto AppUser{ get; set;}
//         public string CityName{ get; set;}
//         public string Name{ get; set;}
//         public string Introduction {get; set;}
//         public string AccessInfo{ get; set;}
//         public double Latitude{ get; set;}
//         public double Longitude{ get; set;}
//         public ICollection<PhotoForReturnDto> Photos { get; set; }
//         public int LikedCount{ get; set;}
//         public int ReviewedCount{ get; set;}
//         public ICollection<AttractionCategory> Categories { get; set; }
//         public string MainPhotoUrl { get; set;}
//         public string WebsiteUrl{ get; set;}
//         public DateTime DateCreated{ get; set;}
//         public DateTime DateUpdated{ get; set;}

File: Wacomi.API/Models/Attraction .cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Wacomi.API.Models
{
    public class Attraction 
    {
        public int Id {get; set;}
        public AppUser AppUser{ get; set;}
        public int? AppUserId { get; set;}
        public City City{ get; set;}
        [Required]
        public int CityId { get; set;}
        [Required]
        [MaxLength(100)]
        public string Name{ get; set;}
        [Required]
        [MaxLength(1500)]
        public string Introduction {get; set;}
        public string AccessInfo{ get; set;}
        public double Latitude{ get; set;}
        public double Longitude{ get; set;}
        public virtual ICollection<Photo> Photos { get; set; }
        public virtual ICollection<AttractionLike> AttractionLikes { get; set;}
        public virtual ICollection<AttractionReview> AttractionReviews { get; set;}
        public virtual ICollection<AttractionCategorization> Categorizations  { get; set; }
        public Photo MainPhoto{ get; set;}
        public int? MainPhotoId{ get; set;}
        public string WebsiteUrl{ get; set;}
        public DateTime DateCreated{ get; set;} = DateTime.Now;
        public DateTime DateUpdated{ get; set;} = DateTime.Now;

    }
}